Read deployment target from `xcconfig` in `Podfile`

mokagio opened the pull request:
This way, we have one less place to change when bumping the iOS version.

## Regression Notes

1. Potential unintended areas of impact – N.A.
2. What I did to test those areas of impact (or what existing automated tests I relied on) – N.A.
3. What automated tests I added (or what prevented me from doing so) – N.A.

---

**PR submission checklist:**

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

**UI changes testing checklist:** Not a UI PR.

mokagio committed:
Read deployment target from `xcconfig` in `Podfile`

This way, we have one less place to change when bumping the iOS version.

jkmassel commented on Podfile:
```diff
@@ -13,7 +14,7 @@ raise 'Please run CocoaPods via `bundle exec`' unless %w[BUNDLE_BIN_PATH BUNDLE_
 use_frameworks!
 
-app_ios_deployment_target = Gem::Version.new('13.0')
+app_ios_deployment_target = Gem::Version.new(Xcodeproj::Config.new(File.new('./config/Common.xcconfig')).to_hash['IPHONEOS_DEPLOYMENT_TARGET'])
```
```suggestion
PROJECT_ROOT_FOLDER = File.expand_path(__dir__)
COMMON_CONFIG_PATH = File.join(PROJECT_ROOT_FOLDER, 'config', 'Common.xcconfig')

app_ios_deployment_target = Gem::Version.new(Xcodeproj::Config.new(COMMON_CONFIG_PATH).to_hash['IPHONEOS_DEPLOYMENT_TARGET'])
```

Just a suggestion that handles pathing properly if `bundle exec pod` isn't run from the root directory.

mokagio replied:
Nice. It didn't occur to me, thanks @jkmassel.

Indeed, the current version run from e.g. `fastlane` fails with file not found:

<img width="1261" alt="image" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/2df64e11-0107-4bd9-bc51-e661aedb6d80">

Using an absolute path makes it robust:

![image](https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/369d23b6-8651-4299-9faf-604bccbab492)

---

I had trouble with absolute paths while working on the Gutenberg local podspec because the resulting hash was different across machines, thus breaking our `Podfile.lock` validation. But I don't think the `Podfile` checksum is done on the interpreted code. Let's see what CI says...

jkmassel reviewed (approved):
I like where you're going with it – left a suggested bug fix; may require updating the hash as well, but then feel free to land!

Disabled auto-merge to allow time to apply the fix

mokagio committed:
Use an absolute path when reading `xcconfig` in `Podfile`

This makes the command executable outside the project's root.

mokagio committed:
Make iOS deployment version a Ruby constant in `Podfile`

Just for consistency with the file path and to write what is hopefully
more idiomatic Ruby.
